http.filesystem.config.server/README.md

```diff
@@ -0,0 +1,156 @@
+- helm 3.0+
+- curl
+
+### Setup
+
+The `setup.sh` script:
+- installs Zilla config server to the Kubernetes cluster with helm and waits for the pod to start up
+- places the contents of the www directory to the Zilla pod
+- starts port forwarding
+- starts a Zilla instance with http echo configuration served by the Zilla config server and waits for the pod to start up
+
+```bash
+$ ./setup.sh
++ helm install zilla-config-server chart --namespace zilla-config-server --create-namespace --wait
+NAME: zilla-config-server
+LAST DEPLOYED: Thu Mar  2 16:26:49 2023
+NAMESPACE: zilla-config-server
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
++ kubectl port-forward --namespace zilla-config-server service/zilla-config 8081 9091
++ nc -z localhost 8081
++ kubectl port-forward --namespace zilla-config-server service/zilla-http 8080 9090
++ sleep 1
++ nc -z localhost 8081
+Connection to localhost port 8081 [tcp/sunproxyadmin] succeeded!
++ nc -z localhost 8080
+Connection to localhost port 8080 [tcp/http-alt] succeeded!
+```
+
+### Verify behavior of the Zilla config server
+
+```bash
+$ curl http://localhost:8081/zilla.yaml
+---
+name: example
+vaults:
+  server:
+    type: filesystem
+    options:
+      keys:
+        store: localhost.p12
+        type: pkcs12
+        password: ${{env.KEYSTORE_PASSWORD}}
+bindings:
+  tcp_server0:
+    type: tcp
+    kind: server
+    options:
+      host: 0.0.0.0
+      port: 8080
+    exit: http_server0
+  tcp_server1:
+    type: tcp
+    kind: server
+    options:
+      host: 0.0.0.0
+      port: 9090
+    exit: tls_server0
+  tls_server0:
+    type: tls
+    kind: server
+    vault: server
+    options:
+      keys:
+        - localhost
+      sni:
+        - localhost
+      alpn:
+        - http/1.1
+        - h2
+    exit: http_server0
+  http_server0:
+    type: http
+    kind: server
+    routes:
+      - when:
+          - headers:
+              :scheme: http
+              :authority: localhost:8080
+              :path: /echo
+          - headers:
+              :scheme: https
+              :authority: localhost:9090
+              :path: /echo
+        exit: echo_server0
+  echo_server0:
+    type: echo
+    kind: server
+```
+The same URL will be used by the Zilla HTTP echo server to query its configuration.
+
+### Verify behavior of the Zilla HTTP echo server
+
+```bash
+$ curl -d "Hello, world" -H "Content-Type: text/plain" -X "POST" http://localhost:8080/echo
+Hello, world
+```
+
+### Change the configuration of the Zilla HTTP echo server
+The Zilla HTTP echo server currently echoes only for the /echo HTTP path. Let's change it to /echo_changed path.
+
+```bash
+$ ./change_config.sh
++ kubectl --namespace zilla-config-server create configmap configmap-served --from-file zilla.yaml -o yaml --dry-run=client
++ kubectl apply -f -
++ sleep 10
++ sleep 10
++ sleep 10
++ sleep 10
++ sleep 10
++ sleep 10
+```
+Note: Kubernetes will not apply the changed configmap immediately, only after ~60 seconds.
```

Comment:
Why does it take ~60secs to apply the change? I'd expect it to be fairly instantaneous.

Reply:
See the answer here: https://kubernetes.io/docs/concepts/configuration/configmap/#mounted-configmaps-are-updated-automatically
I'm trying to come up with a solution to decrease it.

Reply:
I googled a lot. It seems like we'd need to change the syncPeriod in the kubeletConfiguration. That is a kubernetes cluster specific config. I think we're not able to change that in Docker Kubernetes cluster (at least I couldn't find a way to do it).